doc/dev/design/workflows/storage_devices_and_osds.rst

```diff
@@ -0,0 +1,304 @@
+==============================================
+Storage devices and OSD’s management workflows
+==============================================
+The cluster storage devices are the physical storage devices installed in each of the cluster’s hosts. We need to execute different operations over them and also to retrieve information about physical features and working behaviour.
+The basic use cases we have in this area are:
+
+- `1. Retrieve device information. Inventory`_
+- `2. Add OSDs`_
+- `3. Remove OSDs`_
```

Comment:
Typically a `replace` operation is also considered pretty basic for OSDs. Given that the ceph cli has specific commands for that, maybe that should be added here too. Its unfortunately not as simple as a remove, followed by an add

Reply:
Replace and OSD , is just to reuse the ID... this is taking into account in the remove (preserve OSD id)  and in the add operation (reuse osd id).
What about to provide a specific way to replace OSD : selecting at the same time,  one OSD (the one to be deleted) and one available device ( the device to use to create a OSD with the old OSD id) ?

Reply:
In most large scale deployment, there might not be any spare disk in an OSDs node. All disks are included in day1.
So when replacing a disk, normally it's a two-step operation. Pull out the bad one and insert a new one.

Reply:
Added the use case.